crates/goose-server/src/routes/config_management.rs

```diff
@@ -99,6 +112,21 @@ pub async fn upsert_config(
     Json(query): Json<UpsertConfigQuery>,
 ) -> Result<Json<Value>, StatusCode> {
     let config = Config::global();
+
+    // Defensive guard: if this is a secret check and the client sent a boolean 'true' to
+    // indicate presence, do not write that boolean into the secret storage. Treat as a no-op.
+    if query.is_secret {
+        if let Value::Bool(b) = &query.value {
+            if *b {
+                tracing::info!(key = %query.key, "Skipping upsert for secret key with boolean true (presence-only)");
+                return Ok(Json(Value::String(format!(
+                    "Skipped secret upsert for {} (presence-only)",
+                    query.key
+                ))));
+            }
+        }
+    }
```

Comment:
I'm still not convinced that this is the way. can you add a TODO(Douwe) clean this up?